x-pack/plugins/apm/server/routes/service_groups/route.ts

```diff
@@ -39,6 +38,67 @@ const serviceGroupsRoute = createApmServerRoute({
   },
 });
 
+const serviceGroupsWithServiceCountRoute = createApmServerRoute({
+  endpoint: 'GET /internal/apm/service_groups/services_count',
+  params: t.type({
+    query: rangeRt,
+  }),
+  options: {
+    tags: ['access:apm'],
+  },
+  handler: async (
+    resources
+  ): Promise<{ servicesCounts: Record<string, number> }> => {
+    const { context, params } = resources;
+    const {
+      savedObjects: { client: savedObjectsClient },
+      uiSettings: { client: uiSettingsClient },
+    } = await context.core;
+
+    const {
+      query: { start, end },
+    } = params;
+
+    const [setup, maxNumberOfServices] = await Promise.all([
+      setupRequest(resources),
+      uiSettingsClient.get<number>(apmServiceGroupMaxNumberOfServices),
+    ]);
+
+    const serviceGroups = await getServiceGroups({
+      savedObjectsClient,
+    });
+
+    const serviceGroupsWithServiceCount = await Promise.all(
+      serviceGroups.map(
+        async ({
+          id,
+          kuery,
+        }): Promise<{ id: string; servicesCount: number }> => {
+          const servicesCount = await getServicesCounts({
```

Comment:
have you compared performance of a single request w/ filter aggregations vs multiple requests?

Reply:
Good point, I haven't checked filter aggregations, @ogupte have you? If not, we could have a look

Reply:
yeah we should probably fix this since it can get out of hand with many service groups.

Reply:
Some findings on this comparison in a related issue (#141242): https://github.com/elastic/kibana/issues/141242#issuecomment-1255586752